analyzer/pom-analyzer/src/main/java/eu/fasten/analyzer/pomanalyzer/pom/DataExtractor.java

```diff
@@ -71,8 +70,17 @@ public DataExtractor() {
      * @return Link to Maven sources jar file
      */
     public String generateMavenSourcesLink(String groupId, String artifactId, String version) {
-        return this.mavenRepos.get(0) + groupId.replace('.', '/') + "/" + artifactId + "/"
-                + version + "/" + artifactId + "-" + version + "-sources.jar";
+        for (var repo : this.mavenRepos) {
+            var url = repo + groupId.replace('.', '/') + "/" + artifactId + "/"
+                    + version + "/" + artifactId + "-" + version + "-sources.jar";
+            var file = httpGetToFile(url);
```

Comment:
In case of insufficient write permission httpGetToFile(url) will return Option.empty(), while `...-sources.jar` actually exists. Either check existence without downloading the file, or take a look at how error handling is done in OPAL's MavenCoordinate.downloadArtifact() method

Reply:
Done